### ISSUE TITLE
WAR deployed to Tomcat results in PAC-MAN error

### ISSUE THREAD

**Issue by enreas** [jhipster/generator-jhipster#8860]
## Overview of the issue

After generating a production WAR and deploying it to Tomcat, the PAC-MAN loader ends with an error:

![01-loader](https://user-images.githubusercontent.com/1179213/48787247-b6e79c00-ece8-11e8-82ce-39847334753d.png)

There is no information about the error in `catalina.out`:

```
2018-11-20 17:02:27.916  INFO 1480 --- [io-8080-exec-52] org.jhipster.blog.config.WebConfigurer   : Web application configuration, using profiles: prod
2018-11-20 17:02:27.928  INFO 1480 --- [io-8080-exec-52] org.jhipster.blog.config.WebConfigurer   : Web application fully configured
20-Nov-2018 17:02:35.124 INFO [http-nio-8080-exec-52] org.apache.catalina.startup.HostConfig.deployWAR Deployment of web application archive [/var/lib/tomcat8/webapps/blog-0.0.1-SNAPSHOT.war] has finished in [26,175] ms
```

## Reproduce the error

1. Execute `jhipster`.
2. Choose these options:

![02-options](https://user-images.githubusercontent.com/1179213/48787280-c666e500-ece8-11e8-86c3-72588f824ef8.png)

3. Generate a production WAR

    `./gradlew -Pprod -Pswagger bootWar`
    
4. Modify `application-prod.yml` to set database user and password.
5. Create database and user in server.
6. Deploy WAR via Tomcat. App will start, but will show the PAC-MAN error when accessing it.

## JHipster Version(s)

I've noticed this in versions 5.5.0, 5.6.0, 5.6.0, and 5.7.0. Version 5.4.2 is not affected by this issue.

### jhipster info

```
INFO! Using JHipster version installed locally in current project's node_modules
INFO! Executing jhipster:info
INFO! Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
blog@0.0.0 /Users/tests/Desktop/jhipster
└── generator-jhipster@5.7.0

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "org.jhipster.blog",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.7.0",
    "applicationType": "monolith",
    "baseName": "blog",
    "packageName": "org.jhipster.blog",
    "packageFolder": "org/jhipster/blog",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mariadb",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [
      "gatling",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "pt-pt",
      "es"
    ]
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_162"
Java(TM) SE Runtime Environment (build 1.8.0_162-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.162-b12, mixed mode)

git version 2.17.2 (Apple Git-113)

node: v8.12.0

npm: 6.4.1

yeoman: 2.0.2

yarn: 1.12.3

INFO! Congratulations, JHipster execution is complete!
```

## Browsers and Operating System

Browser: Safari 12.0.1
Operating System: Ubuntu 18.04.1 LTS

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
Did you follow everything mentioned in the error message? like building
your front-end?

Thanks & Regards,
Deepu

**Comment by enreas** [comment]
Yes, I built it using `./gradlew -Pprod -Pswagger bootWar`.

**Comment by fleboulch** [comment]
Which war do you deploy on tomcat?
This issue has probably already been mentioned in #8811.

@enreas could you confirm you are facing the same issue?

**Comment by atomfrede** [comment]
As gradle currently does *not* generate a `.original` war file the deployment via tomcat will most properly fail. @enreas can you try to edit your `prod.gradle` file with the mentioned changes regarding the war file (https://github.com/jhipster/generator-jhipster/issues/8811#issuecomment-438155721)?
Afterward you still need to trigger the `war` build via `bootWar` or `war` task in case there is no war with `original` in its name generated.

**Comment by atomfrede** [comment]
You are right in my local test setup I also added it to main build file.

**Comment by enreas** [comment]
Please let me confirm this is solved in 5.7.1. And let me also thank you for your help and time.

**Comment by jdubois** [comment]
Thanks a lot @enreas for your feedback!
